Mcode/+logger/raiseExceptions.m

```diff
@@ -0,0 +1,46 @@
+function varargout = raiseExceptions(newval)
+%RAISEEXCEPTIONS gets/sets a flag for handling exceptions during logging
+%   RAISEEXCEPTIONS gets or sets a flag that is used by the logic that
+%   controls the handling of exceptions raised/thrown during logging.
+%   The default flag value is true.
+%
+%   Usage
+%       % suppress exceptions during logging framework message handling
+%       logger.raiseExceptions(false)
+
+%{
+This mechanism is inspired by a comment from Martijn Pieters in
+https://stackoverflow.com/questions/66587941/what-happens-if-a-python-logging-handler-raises-an-exception
+
+"The Python standard-library handlers have been build with robustness in 
+mind.  If an exception is raised ..., all standard library 
+implementations catch the exception and ...
+By default, ... re-raises the exception.  In production systems you want 
+to set logging.raiseExceptions = False, at which point the exceptions 
+are silently ignored, and logging continues as if nothing happened.
+
+From the documentation: ... If the module-level attribute 
+raiseExceptions is False, exceptions get silently ignored. This is what 
+is mostly wanted for a logging system - most users will not care about 
+errors in the logging system, they are more interested in application 
+errors.  ... (The default value of raiseExceptions is True, as that is 
+more useful during development)."
+%}
+
+persistent flag
```

Comment:
Let's back the flag value with appdata, so that its state survives a `clear classes` or other action that clears the raiseException function definition. Something like this:

```
function out = raiseExceptions(newval)

persistent flag
if isempty(flag)
  flag = getappdata(0, 'SLF4M_raiseExceptions');
  if isempty(flag)
    flag = true;
  end
end

if nargin == 0
  out = flag;
else
  if ~(islogical(newval) && isscalar(newval))
    error('The raiseExceptions value must be a scalar logical; got a %s %s', mat2str(size(newval)), class(newval));
  end
  flag = newval;
  setappdata(0, 'SLF4M_raiseExceptions', flag);
end

end
```

(Oops: I was going to say that SLF4M's code base uses 2-space indents, but it looks like my code actually has a mix of 2-space or 4-space indents, on a per-file basis. And the `.editorconfig` file says it's 4 spaces. I need to pick one and make it all uniform. https://github.com/janklab/SLF4M/issues/13)